pkg/network/cache/cache_suite_test.go

```diff
@@ -1,11 +1,33 @@
 package cache_test
 
 import (
+	"io/ioutil"
+	"sync"
 	"testing"
 
+	"kubevirt.io/kubevirt/pkg/network/cache"
+
+	kfs "kubevirt.io/kubevirt/pkg/os/fs"
+
 	"kubevirt.io/client-go/testutils"
 )
 
 func TestCache(t *testing.T) {
 	testutils.KubeVirtTestSuiteSetup(t)
 }
+
+type tempCacheCreator struct {
+	once   sync.Once
+	tmpDir string
+}
+
+func (c *tempCacheCreator) New(filePath string) *cache.Cache {
+	c.once.Do(func() {
+		tmpDir, err := ioutil.TempDir("", "temp-cache")
+		if err != nil {
+			panic("Unable to create temp cache directory")
+		}
+		c.tmpDir = tmpDir
+	})
+	return cache.NewCustomCache(filePath, kfs.NewWithRootPath(c.tmpDir))
+}
```

Comment:
Maybe we can use the testing TempDir feature so we don't have the `once` sync

https://pkg.go.dev/testing#B.TempDir

Reply:
I do not think it will solve it.
We need one, and only one place the files are written/read from for all instances that are created from this creator.

We are actually limited here by how the cache is called by its users. Instead of having a cache object per VMI, which is used throughout the setup flow, we need to create these again and again.
We will change this with the coming refactorings, but for now we are stuck with this as is.